### PR TITLE
[JENKINS-28058] Don't send a reference to Computer.class over remoting channels

### PR DESCRIPTION
**Pull request by stephenc**
[JENKINS-28058](https://issues.jenkins-ci.org/browse/JENKINS-28058)

@reviewbybees

### CODE REVIEW

**Comment by jtnord** [comment]
:-1: all I see is logging changes - nothing related to the description of the checking.

**Comment by jtnord** [comment]
Oh.. it needed the Computers Logger....

:+1:

**Comment by jglick** [comment]
`UpdateCenter2Test.install` failure is unrelated.

File: core/src/main/java/hudson/model/Computer.java [hunk 1]
```diff
@@ -1191,27 +1194,29 @@ public String getHostName() throws IOException, InterruptedException {
     }
 
     private static class ListPossibleNames extends MasterToSlaveCallable<List<String>,IOException> {
+        private static final Logger LOGGER = Logger.getLogger(ListPossibleNames.class.getName());
```
Comment (jglick) [inline core/src/main/java/hudson/model/Computer.java]: Generally would use the top-level `Computer.class`.

**Comment by jglick** [comment]
-0 since there is no test demonstrating the fix and guarding against regressions, which seem quite likely to appear from innocent-looking changes.

More broadly, I think the slave class loader should have an actual blacklist. This is not just a performance problem; [JENKINS-16920](https://issues.jenkins-ci.org/browse/JENKINS-16920) and similar issues arise from the slave mistakenly receiving classes which it should not have been using.

**Comment by stephenc** [comment]
@jglick can you suggest how to test?

**Comment by jglick** [comment]
> can you suggest how to test?

Well, if there is some logging (even at `FINER` or whatever) from `RemoteClassLoader` saying what it is loading, you can have the test install a `Handler` capturing those messages. (Confusingly, you must `setLevel` on _both_ the `Handler` and the `Logger`.)

But again my preference would be to have the class loader actually throw `IllegalStateException` or the like if an attempt is made to load a bogus class like `Jenkins` from the slave side, in which case no special test is likely to be needed: existing tests would begin failing until you did the main sources change.

**Comment by stephenc** [comment]
@jglick I'm not seeing any reliable log message to catch, and it really needs @kohsuke to add the block on loading `Jenkins` but in any case I'm not seeing a specific test for this that we can add

**Comment by jglick** [comment]
Well +0 as it stands: harmless, might be effective for a while.

**Pull request merged**